### ISSUE TITLE
If you resharing with a third user (user3) via Web Server and later on the Desktop client you go to add other permission the application does not work

### ISSUE THREAD

**Issue by Dianafg76** [owncloud/client#4358]
... related https://github.com/owncloud/client/issues/4357
If you resharing with a third user (user3) via Web Server and later:
### Steps to reproduce
1. Go to the Desktop client with user2
2. Go to the folder (Share with user2)
3. Right click and Share with owncloud
![screen shot 2016-01-14 at 14 13 49](https://cloud.githubusercontent.com/assets/12032058/12325245/1a3a81f0-bac9-11e5-8198-efa348a51ea4.png)
4. If you click on Can edit

### Actual behaviour
If you resharing with a third user (user3) via Web Server and later on the Desktop client you go to add other permission the application does not work

Appears this error on the LOG:
```
01-14 14:05:03:915 0x7ff8c0d1af10 OCC::AbstractNetworkJob::start: !!! OCC::OcsShareJob created for "https://xxxxxxxxxxxxxxxx” + "ocs/v1.php/apps/files_sharing/api/v1/shares/12" ""
01-14 14:05:04:246 0x7ff8c0d1af10 OCC::OcsJob::finished: Reply to "PUT"  QUrl( "https://xxxxxxxx/ocs/v1.php/apps/files_sharing/api/v1/shares/12" )  (QPair("permissions","31")) has unexpected status code: 404 "{"ocs":{"meta":{"status":"failure","statuscode":404,"message":"Setting permissions for 432 failed, because the permissions exceed permissions granted to di"},"data":[]}}"
```
![screen shot 2016-01-14 at 14 14 52](https://cloud.githubusercontent.com/assets/12032058/12325307/9a70ed3c-bac9-11e5-86de-163b50c7b664.png)

### Server configuration
Desktop v ownCloud-2.1.1.3024-nightly20160112.pkg
Server v {"installed":true,"maintenance":false,"version":"8.2.2.2","versionstring":"8.2.2","edition":"Enterprise"}

cc/ @davivel

**Comment by ckamm** [comment]
The problem seems to be that when we create the share with `PermissionDefault`, we don't send any permission information to the server. As a response we get `Sharing <N> failed, because the permissions exceed permissions granted to <user>`.

It seems like the server assumes we want to reshare with permissions we don't have.

As a workaround, we could detect the permissions we were granted and create the new share with those.

**Comment by ckamm** [comment]
I don't think we can get at the permissions for something that was shared with us. For example, sharing a folder with 'change' permission, just adds 'W' permissions on the contained files and is not reflected in the folder permissions themselves, I think. @rullzer ?

So the workaround probably is to try creating a share with minimal permissions if it fails to be created with default permissions.

**Comment by rullzer** [comment]
@ckamm yeah by default we pass no permissions. So that the default is used. However if the permissions granted to us are less than the default there is an error.

The main issue is that the oc:permission field is different from the share permissions. Assume a directory shared with us. That has delete permissions (since we can unshare it from ourself). But this does not tell us if we have delete permissions on the share.

Creating a minimal share might be a solution. But then the sharing behaviour is different from what we have on the server and might confuse a lot of users.

A approach that *should* work is the `shares?shared_with_me=true` command. This will return all the shares shared with the current user (with permissions). However this does not really scale on large installations requesting this every time.

In other words we need an API change. Which I alrady proposed in https://github.com/owncloud/core/issues/21798 but that won't happen server side for 9.0

**Comment by ogoffart** [comment]
The reason we don't send permission to the server is because of https://github.com/owncloud/client/issues/4152   
We want to have the same default as the server.  And the server decides which default.

**Comment by ckamm** [comment]
@ogoffart It turns out the server can selects permission default that are impossible to grant. Do you see any problems with figuring out the maximum possible permissions and then explicitly using these?

**Comment by Dianafg76** [comment]
If you resharing with a third user (user3) via Web Server and later on the Desktop client you go to add other permission the application does not work
Desktop v 2.2.0 build 3111
Server v {"installed":true,"maintenance":false,"version":"8.2.2.2","versionstring":"8.2.2","edition":"Enterprise"}

**Comment by rullzer** [comment]
As far as I can tell this is now fixed with https://github.com/owncloud/client/pull/4501
